### PR TITLE
Minor fixes in docs, code and examples to reduce CI pipeline errors

### PR DESCRIPTION
**Pull request by andres-fr**
The PR proposed in https://github.com/f-dangel/curvlinops/pull/208 had 3 main lines of contribution:

    Updating Python to 3.10, since 3.9 is deprecated
    General CI fixes
    The actual content of the PR

It was agreed to split those PR into 3 subsequent, individual PRs. 
The first one has already been merged to main.
The present PR is the second one of the bunch, i.e. "general CI fixes"

All modifications in this PR have the aim to prevent a linter or autodoc from crashing and should not create any changes in behaviour or semantics.

### CODE REVIEW

File: docs/examples/basic_usage/example_inverses.py [hunk 1]
```diff
@@ -271,22 +275,25 @@
...
+for i, (n, inv) in enumerate(zip(num_terms, neumann_inverses_cpu)):
     ax = axes.flat[i]
     ax.set_title(f"Neumann, {n} terms")
     image = ax.imshow(inv.abs().log10())
     plt.colorbar(image, ax=ax, shrink=0.5)
 
 ax = axes.flat[-1]
 ax.set_title("Exact inverse")
-image = ax.imshow(inv_damped_GGN_mat.abs().log10())
+image = ax.imshow(inv_damped_GGN_mat_cpu.abs().log10())
```
Comment (f-dangel) [inline docs/examples/basic_usage/example_inverses.py]: Same here. No need to introduce a new list, simply add `.detach().cpu()` inside the plotting commands.

**Review by f-dangel — CHANGES_REQUESTED** [review]
Some minor comments. I think we can merge after one iteration.
